iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -621,8 +617,13 @@ private void commitCompaction(Table table, long startTime, FilesForCommit result
       LOG.debug("Compacted full table with files {}", results);
     } else {
       // Single partition compaction
-      List<DataFile> existingDataFiles = IcebergTableUtil.getDataFiles(table, partitionSpecId, partitionPath);
-      List<DeleteFile> existingDeleteFiles = IcebergTableUtil.getDeleteFiles(table, partitionSpecId, partitionPath);
+      List<DataFile> existingDataFiles = partitionPath != null ?
```

Comment:
i feel that this became overloaded, could we extract this into `CompactionOperation` 
see examples in iceberg: 
1. CopyOnWriteOperation extends BaseBatchWrite
2. DynamicOverwrite extends BaseBatchWrite

Reply:
I added one more parameter to the function and got rid of the overloaded functions, is it fine now?